src/config/reader/reader.rs

```diff
@@ -145,18 +147,126 @@ impl ConfigReader {
                         .push(Content { id: Some(id), content: file_descriptor_set });
                 }
                 LinkType::Script => {
+                    let path = Self::resolve_path(&config_link.src, parent_dir);
+                    let source = self.read_file(&path).await?;
+                    let content = source.content;
+
                     config_module.extensions.script = Some(content);
                 }
                 LinkType::Cert => {
+                    let path = Self::resolve_path(&config_link.src, parent_dir);
+                    let source = self.read_file(&path).await?;
+                    let content = source.content;
+
                     config_module
                         .extensions
                         .cert
                         .extend(self.load_cert(content.clone()).await?);
                 }
                 LinkType::Key => {
+                    let path = Self::resolve_path(&config_link.src, parent_dir);
+                    let source = self.read_file(&path).await?;
+                    let content = source.content;
+
                     config_module.extensions.keys =
                         Arc::new(self.load_private_key(content.clone()).await?)
                 }
+                LinkType::ReflectionWithFileName => {
+                    let mut file_descriptor_set = FileDescriptorSet::default();
+
+                    let file_descriptor_proto = get_by_proto_name(
+                        config_link.src.as_str(),
+                        &self.runtime,
+                        &config_link
+                            .id
+                            .clone()
+                            .context("Expected link id with method name but found none")?,
```

Comment:
this error is showed to user in case of errors but in such form it says more about the code and semantics not about the config and expected actions.

Consider updating errors message accross pr with clear messages from user perspective and possible actions to fix. Also trace for definition could be helpful.
i.e. "id for link is not specified. When using grpc ... it is mandatory"

Reply:
let's see if we continue with `Reflection*` or switch to single `GrpcReflection`. I'll update the errors later